### PR TITLE
Algorithms 42 to 50 updated with Histogram Data

### LINKED ISSUE #17678: HistogramData rollout: algorithms 42-50
HistogramData rollout: algorithms

This issue is part of the HistogramData rollout, see #17641.
Algorithms covered in this issue are:

- Framework/Algorithms/src/StripVanadiumPeaks.cpp
- Framework/Algorithms/src/SumEventsByLogValue.cpp
- Framework/Algorithms/src/SumRowColumn.cpp
- Framework/Algorithms/src/SumSpectra.cpp
- Framework/Algorithms/src/TOFSANSResolutionByPixel.cpp
- Framework/Algorithms/src/TOFSANSResolution.cpp
- Framework/Algorithms/src/Transpose.cpp
- Framework/Algorithms/src/UnaryOperation.cpp

### PR DESCRIPTION
**Pull request by dtasev**
HistogramData rollout: algorithms

This issue is part of the HistogramData rollout, see #17641.
Algorithms covered in this issue are:

```
Framework/Algorithms/src/StripVanadiumPeaks.cpp
Framework/Algorithms/src/SumEventsByLogValue.cpp
Framework/Algorithms/src/SumRowColumn.cpp
Framework/Algorithms/src/SumSpectra.cpp
Framework/Algorithms/src/TOFSANSResolutionByPixel.cpp
Framework/Algorithms/src/TOFSANSResolution.cpp
Framework/Algorithms/src/Transpose.cpp
Framework/Algorithms/src/UnaryOperation.cpp
```

Fixed #17678

### CODE REVIEW

File: Framework/Algorithms/src/He3TubeEfficiency.cpp [hunk 1]
```diff
@@ -171,7 +171,7 @@ void He3TubeEfficiency::correctForEfficiency(std::size_t spectraIndex) {
 
   Mantid::MantidVec &yout = this->outputWS->dataY(spectraIndex);
   Mantid::MantidVec &eout = this->outputWS->dataE(spectraIndex);
-  // Need the original values so this is not a reference
+  // Need the original values so this is not a referencegit
```
Comment (SimonHeybrock) [inline Framework/Algorithms/src/He3TubeEfficiency.cpp]: typo

File: Framework/Algorithms/src/StripVanadiumPeaks.cpp [hunk 2]
```diff
@@ -87,10 +87,11 @@ void StripVanadiumPeaks::exec() {
...
   const int nhists = static_cast<int>(inputWS->getNumberHistograms());
   Progress progress(this, 0.0, 1.0, nhists * 2);
+
   for (int k = 0; k < nhists; ++k) {
-    outputWS->dataX(k) = inputWS->readX(k);
-    outputWS->dataY(k) = inputWS->readY(k);
-    outputWS->dataE(k) = inputWS->readE(k);
+    outputWS->setSharedX(k, inputWS->sharedX(k));
+    outputWS->setSharedY(k, inputWS->sharedY(k));
+    outputWS->setSharedE(k, inputWS->sharedE(k));
```
Comment (SimonHeybrock) [inline Framework/Algorithms/src/StripVanadiumPeaks.cpp]: `outputWS->setHistogram(k, inputWS->histogram(k));`

File: Framework/Algorithms/src/StripVanadiumPeaks.cpp [hunk 3]
```diff
@@ -114,18 +115,20 @@ void StripVanadiumPeaks::exec() {
...
-      MantidVec Y = outputWS->dataY(k);
+      auto &X = outputWS->mutableX(k);
+      auto &Y = outputWS->mutableY(k);
 
+      // clear previous data, should be fast O(1) because primitive type
+      midX.clear();
       // Middle of each X bin
-      MantidVec midX;
-      convertToBinCentre(X, midX);
+      convertToBinCentre(X.rawData(), midX);
```
Comment (SimonHeybrock) [inline Framework/Algorithms/src/StripVanadiumPeaks.cpp]: Just use `auto midX = outputWS->points(k)`.

File: Framework/Algorithms/src/StripVanadiumPeaks.cpp [hunk 4]
```diff
@@ -114,18 +115,20 @@ void StripVanadiumPeaks::exec() {
   // Get the width percentage
   double widthPercent = getProperty("PeakWidthPercent");
 
+  std::vector<double> midX(nhists);
   for (int k = 0; k < nhists; ++k) {
     if ((!singleSpectrum) || (singleIndex == k)) {
       // Get the X and Y vectors
-      MantidVec X = outputWS->dataX(k);
-      MantidVec Y = outputWS->dataY(k);
+      auto &X = outputWS->mutableX(k);
```
Comment (SimonHeybrock) [inline Framework/Algorithms/src/StripVanadiumPeaks.cpp]: Why mutable?

File: Framework/Algorithms/src/SumSpectra.cpp [hunk 5]
```diff
@@ -147,7 +147,7 @@ void SumSpectra::exec() {
     auto &outSpec = outputWorkspace->getSpectrum(0);
 
     // Copy over the bin boundaries
-    outSpec.dataX() = localworkspace->readX(0);
+    outSpec.mutableX() = localworkspace->x(0);
```
Comment (SimonHeybrock) [inline Framework/Algorithms/src/SumSpectra.cpp]: Share?

File: Framework/Algorithms/src/SumSpectra.cpp [hunk 6]
```diff
@@ -479,7 +479,7 @@ void SumSpectra::execEvent(EventWorkspace_const_sptr localworkspace,
   }
 
   // Set all X bins on the output
-  outputWorkspace->setAllX(HistogramData::BinEdges(localworkspace->refX(0)));
+  outputWorkspace->setAllX(HistogramData::BinEdges(localworkspace->sharedX(0)));
```
Comment (SimonHeybrock) [inline Framework/Algorithms/src/SumSpectra.cpp]: `outputWorkspace->setAllX(localworkspace->binEdges(0));`

File: Framework/Algorithms/src/Transpose.cpp [hunk 7]
```diff
@@ -39,38 +39,49 @@ void Transpose::exec() {
...
   auto newXVector =
       Kernel::make_cow<HistogramData::HistogramX>(std::move(newXValues));
 
+  // branch out
+  // if outRebinWorkspace
+  // if(outRebinWorkspace){
+  //   // do stuff
+  // }else{ // else, not rebin ws
+
+  // }
```
Comment (SimonHeybrock) [inline Framework/Algorithms/src/Transpose.cpp]: WIP?

**Review by SimonHeybrock — COMMENTED** [review]
Looks good apart from minor comments :+1:

**Comment by DimitarTasev** [comment]
Restarted Windows, unrelated test failure on `MantidPlotProjectSerialiseTest` http://builds.mantidproject.org/job/pull_requests-win7/10443/console

**Review by SimonHeybrock — COMMENTED** [review]
Just 2 small comments, apart from that I think this is ready:

**Comment by DimitarTasev** [comment]
Restarted OSX due to unrelated unit test failure http://builds.mantidproject.org/job/pull_requests-osx/9846/

**Comment by SimonHeybrock** [comment]
:shipit:

**Pull request merged**